sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -716,22 +717,38 @@ ProgramManager::collectDependentDeviceImagesForVirtualFunctions(
       // virtual-functions-set properties, but their handling is the same: we
       // just grab all sets they reference and add them for consideration if
       // we haven't done so already.
+      bool isDummyImage = false;
       for (const sycl_device_binary_property &VFProp :
            BinImage->getVirtualFunctions()) {
-        std::string StrValue = DeviceBinaryProperty(VFProp).asCString();
+        if (VFProp->Name == std::string_view("dummy-image")) {
```

Comment:
FYI: you can use `sv` string literal to make the code less verbose: [cppreference](https://en.cppreference.com/w/cpp/string/basic_string_view/operator%22%22sv)